src/main/java/com/github/manolo8/darkbot/gui/titlebar/BackpageTask.java

```diff
@@ -0,0 +1,264 @@
+                else {
+                    File file = to.toFile();
+                    try (BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(file))) {
+                        double compressionRatio = (double) entry.getCompressedSize() / entry.getSize();
+
+                        int read;
+                        byte[] buffer = new byte[8192];
+                        while ((read = zis.read(buffer)) != -1) {
+                            bos.write(buffer, 0, read);
+                            setProgress(progressBar, downloadedBytes += (read * compressionRatio));
+                        }
+
+                        if (OSUtil.isLinux() && file.getName().startsWith(EXECUTABLE_NAME))
+                            file.setExecutable(true);
+                    }
+                }
+            }
+        }
+    }
+
+    private void setProgress(JProgressBar progressBar, int progress) {
+        SwingUtilities.invokeLater(() -> progressBar.setValue(progress));
+    }
+
+    private @Nullable Version readVersionFile() throws IOException {
+        if (Files.notExists(VERSION_PATH)) return null;
+        return new Version(Files.readString(VERSION_PATH));
+    }
+
+    private void writeVersionFile(Version version) throws IOException {
+        Files.writeString(VERSION_PATH, version.toString());
+    }
+
+    public static class ReleaseInfo {
+        public String name;
+        public List<Asset> assets;
+
+        public static ReleaseInfo get() throws IOException, JsonParseException {
+            return new Gson().fromJson(Http.create(RELEASE_URL).getContent(), ReleaseInfo.class);
+        }
+
+        public Version getVersion() {
+            return new Version(name);
+        }
+
+        public Asset getValidAsset() {
+            return assets.stream()
+                    .filter(ReleaseInfo.Asset::isSupported)
+                    .findFirst().orElse(null);
+        }
+
+        public static class Asset {
+            @SerializedName("browser_download_url")
+            public String downloadUrl;
+            public String state;
+            public int size;
+
+            public InputStream openStream() throws IOException {
+                return new URL(downloadUrl).openStream();
+            }
+
+            public boolean isSupported() {
+                if (state == null || !state.equals("uploaded")) return false;
+
+                switch (OSUtil.getCurrentOs()) { // maybe should check architecture?
+                    case WINDOWS:
+                        if (downloadUrl.contains("win-x64.zip")) return true;
+                        break;
+                    case MACOS:
+                        if (downloadUrl.contains("mac-x64.zip")) return true;
+                        break;
+                    case LINUX:
+                        if (downloadUrl.contains("linux-x64.zip")) return true;
+                        break;
+                }
+
+                return false;
```

Comment:
Consider adding an OSUtil.getShortName() that returns win, mac, or linux, then this can be simplified:
```suggestion
                return downloadUrl.contains(OSUtil.getShortName() + "-x64.zip");
```